basic_suite_test.go

```diff
@@ -48,23 +53,85 @@ func (s *BasicTestsSuite) TearDownSuite() {
 }
 
+func (s *BasicTestsSuite) TestNSE_CanRegisterInRegistry() {
+	defer require.NoRestarts(s.T())
+
+	s.Require().NoError(exechelper.Run("kubectl apply -f ./deployments/registry-service.yaml", s.options...))
+	s.Require().NoError(exechelper.Run("kubectl apply -f ./deployments/registry-memory.yaml", s.options...))
+	s.Require().NoError(exechelper.Run("kubectl wait --timeout=120s  --for=condition=ready pod -l app=nsm-registry", s.options...))
+
+	s.Require().NoError(exechelper.Run("kubectl apply -f ./deployments/nse.yaml", s.options...))
+	s.Require().NoError(exechelper.Run("kubectl wait --timeout=120s  --for=condition=ready pod -l app=nse", s.options...))
+}
+
+func (s *BasicTestsSuite) TestNSMgr_CanCanFindNSEInRegistry() {
+	defer require.NoRestarts(s.T())
+
+	s.Require().NoError(exechelper.Run("kubectl apply -f ./deployments/registry-service.yaml", s.options...))
+	s.Require().NoError(exechelper.Run("kubectl apply -f ./deployments/registry-memory.yaml", s.options...))
+	s.Require().NoError(exechelper.Run("kubectl wait --timeout=120s  --for=condition=ready pod -l app=nsm-registry", s.options...))
+
+	s.Require().NoError(exechelper.Run("kubectl apply -f ./deployments/nse.yaml", s.options...))
+	s.Require().NoError(exechelper.Run("kubectl wait --timeout=120s  --for=condition=ready pod -l app=nse", s.options...))
+
+	s.Require().NoError(exechelper.Run("kubectl apply -f ./deployments/fake-nsmgr.yaml", s.options...))
+	s.Require().NoError(exechelper.Run("kubectl wait --timeout=120s  --for=condition=ready pod -l app=fake-nsmgr", s.options...))
+}
+
+func (s *BasicTestsSuite) TestProxyRegistryDNS_CanCanFindNSE_Local() {
+	defer require.NoRestarts(s.T())
+
+	s.Require().NoError(exechelper.Run("kubectl apply -f ./deployments/proxy-registry-service.yaml", s.options...))
+	s.Require().NoError(exechelper.Run("kubectl apply -f ./deployments/registry-service.yaml", s.options...))
+
+	s.Require().NoError(k8s.ApplyDeployment("./deployments/registry-proxy-dns.yaml", func(proxyRegistry *v1.Deployment) {
+		proxyRegistry.Spec.Template.Spec.Containers[0].Env = append(proxyRegistry.Spec.Template.Spec.Containers[0].Env, v1core.EnvVar{
+			Name:  "REGISTRY-PROXY-DNS_DOMAIN",
+			Value: "default.svc.cluster.local",
+		})
+	}))
+
+	s.Require().NoError(exechelper.Run("kubectl apply -f ./deployments/registry-memory.yaml", s.options...))
+	s.Require().NoError(exechelper.Run("kubectl wait --timeout=120s  --for=condition=ready pod -l app=nsm-registry", s.options...))
+
+	s.Require().NoError(exechelper.Run("kubectl apply -f ./deployments/nse.yaml", s.options...))
+	s.Require().NoError(exechelper.Run("kubectl wait --timeout=120s  --for=condition=ready pod -l app=nse", s.options...))
+
+	s.Require().NoError(exechelper.Run("kubectl wait --timeout=120s  --for=condition=ready pod -l app=nsm-registry-proxy-dns", s.options...))
+
+	s.Require().NoError(k8s.ApplyDeployment("./deployments/fake-nsmgr.yaml", func(nsmgr *v1.Deployment) {
```

Comment:
Curious why the k8s api here for deployment rather than kubectl ?

Reply:
As I know with yaml models we can't just via kubectl dynamically edit properties.

Reply:
Probably we can prepare yaml models per tests as separate files

Reply:
As an alternative we can switch to using helm, then we'll able to use setvariavles in cmd 
@edwarnicke Thoughts?

Reply:
I guess the question is: why do we need to edit this yaml file?  Why isn't it good to go out of the box?